### PR TITLE
handle None timeout Fixes #274

### LINKED ISSUE #274: TypeError: '<=' not supported between instances of 'NoneType' and 'int', when using run_in_thread sleep_time=None
when using:

```
import fakeredis


def main():
    redis = fakeredis.FakeStrictRedis()
    redis_pubsub = redis.pubsub()
    redis_pubsub.subscribe(my_channel=lambda: None)
    return redis_pubsub.run_in_thread(sleep_time=None)


if __name__ == "__main__":
    main()
```

```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
>>>     self.run()
  File "/home/graingert/.virtualenvs/fr-demo/lib/python3.8/site-packages/redis/client.py", line 2598, in run
    pubsub.get_message(ignore_subscribe_messages=True,
  File "/home/graingert/.virtualenvs/fr-demo/lib/python3.8/site-packages/redis/client.py", line 2513, in get_message
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/graingert/.virtualenvs/fr-demo/lib/python3.8/site-packages/redis/client.py", line 2428, in parse_response
    if not block and not connection.can_read(timeout=timeout):
  File "/home/graingert/.virtualenvs/fr-demo/lib/python3.8/site-packages/fakeredis/_server.py", line 2549, in can_read
    return self._selector.check_can_read(timeout)
  File "/home/graingert/.virtualenvs/fr-demo/lib/python3.8/site-packages/fakeredis/_server.py", line 2474, in check_can_read
    if timeout <= 0:
TypeError: '<=' not supported between instances of 'NoneType' and 'int'

KeyboardInterrupt
```

### PR DESCRIPTION
**Pull request by graingert**
handle None timeout Fixes #274

### CODE REVIEW

File: fakeredis/_server.py [hunk 1]
```diff
@@ -2547,7 +2547,7 @@ class FakeSelector(BaseSelector):
     def check_can_read(self, timeout):
         if self.sock.responses.qsize():
             return True
-        if timeout <= 0:
+        if timeout is None or timeout <= 0:
```
Comment (bmerry) [inline fakeredis/_server.py:2550]: This doesn't seem quite right: from what I can tell, `timeout=None` in redis is meant to mean "infinity", not "0".

Comment (graingert) [reply fakeredis/_server.py:2550]: The test case I have is tricky, as it seems to host hang the thread

**Comment by mgetka** [comment]
IMHO, to mock indefinite timeout, it should rather go like this
```python
class FakeSelector(BaseSelector):
    def check_can_read(self, timeout):
        if self.sock.responses.qsize():
            return True
        if timeout is not None and timeout <= 0:
            return False

        # A sleep/poll loop is easier to mock out than messing with condition
        # variables.
        start = time.time()
        while True:
            if self.sock.responses.qsize():
                return True
            time.sleep(0.01)
            now = time.time()
            if timeout is not None and now > start + timeout:
                return False

    def check_is_ready_for_command(self, timeout):
        return True
```

**Comment by graingert** [comment]
@mgetka sorry I don't have time to fix up this PR, can you take it over?

**Pull request closed without merging**